src/json.cxx

```diff
@@ -15,3 +16,69 @@ std::string json_to_string(rapidjson::Document const &jd) {
   jd.Accept(w);
   return b1.GetString();
 }
+
+/// Merge the json structure v2 into a copy of v1.
+/// If the new key in v2 is not a object, it replaces the key in v1.
+/// Elements of arrays are also not merged, but the new array fully replaces
+/// the old one.
+rapidjson::Document merge(rapidjson::Value const &v1,
+                          rapidjson::Value const &v2) {
+  using namespace rapidjson;
+  using std::string;
+  using std::deque;
+  Document ret;
+  auto &a = ret.GetAllocator();
+  ret.CopyFrom(v1, a);
+  if (!v1.IsObject() || !v2.IsObject()) {
+    return ret;
+  }
+  struct to_copy_element {
+    string name;
+    Value const *v;
+  };
+  struct stack_element {
```

Comment:
This looks good to me. I think you've made the right choice of container (`deque`) and constraining these structs to the scope of the method.